### DISCUSSION TITLE
数据库文件突然变小

### DISCUSSION THREAD

**Discussion by liwenlong68** [apache/iotdb#11000] [Q&A] [answered]
你好
使用版本:0.12.4
我们做了一次数据迁移，直接将Iotdb/data下的文件共120GB全部拷贝到新的服务器对应的iotdb目录中  并在新的服务器中成功启动iotdb且在新的里面能查到老数据
在后面我们继续使用新的iotdb时发现data目录直接减少至60GB，进入data目录里面发现是.tsfile.merge文件从60GB直接减少至1KB不到
我们很疑惑，也很担心数据因此丢失。请大神帮忙解惑
![mmexport1693396908467](https://github.com/apache/iotdb/assets/143609467/597b5c17-79d7-4dbe-abd1-4feec163d34f)
![mmexport1693396906758](https://github.com/apache/iotdb/assets/143609467/88c6d05e-6b16-404e-995c-398d0d0a2823)

**Comment by HTHou** [comment, marked as answer]
tsfile.merge 是合并的中间态文件，合并后会生成正常的tsfile。看起来同名的文件已经正常合并出来了，数据都存在于同名的 tsfile 中。这个 tsfile.merge 文件已经没有用了，iotdb 会正常清理 tsfile.merge 文件。

**Comment by myysophia** [comment]
> 你好 使用版本:0.12.4 我们做了一次数据迁移，直接将Iotdb/data下的文件共120GB全部拷贝到新的服务器对应的iotdb目录中 并在新的服务器中成功启动iotdb且在新的里面能查到老数据 在后面我们继续使用新的iotdb时发现data目录直接减少至60GB，进入data目录里面发现是.tsfile.merge文件从60GB直接减少至1KB不到 我们很疑惑，也很担心数据因此丢失。请大神帮忙解惑 ![mmexport1693396908467](https://private-user-images.githubusercontent.com/143609467/264337450-597b5c17-79d7-4dbe-abd1-4feec163d34f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ3MDc0ODksIm5iZiI6MTcxNDcwNzE4OSwicGF0aCI6Ii8xNDM2MDk0NjcvMjY0MzM3NDUwLTU5N2I1YzE3LTc5ZDctNGRiZS1hYmQxLTRmZWVjMTYzZDM0Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwM1QwMzMzMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNWNiYTY4ZTQ3ZDM5Yjc0NDlkNmIxZjIzZDc4OWEwYzUxMWRkMWM4NTI0M2U3YzkyNjgzZDI0ZDVkMTg2ZTdmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.PU-tkm_gJ7M292OOKnesnVbKUcP4-mbNHkDzNkt6MnU) ![mmexport1693396906758](https://private-user-images.githubusercontent.com/143609467/264337467-88c6d05e-6b16-404e-995c-398d0d0a2823.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ3MDc0ODksIm5iZiI6MTcxNDcwNzE4OSwicGF0aCI6Ii8xNDM2MDk0NjcvMjY0MzM3NDY3LTg4YzZkMDVlLTZiMTYtNDA0ZS05OTVjLTM5OGQwZDBhMjgyMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwM1QwMzMzMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMGRjYzJlZDE2MTY3MzgzMTYxZGJkNTRlNDczMzE3YTNkMTJlYzBhMzU3ODY5MmFlYWQ0NWQ1NTFkY2Q0ZTY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1T-B2a3qcBNMuTUxWv_6lRTLbxROb9Wnk1QG5YiOvCQ)

在物理备份数据目录之前是不是需要将这些中间态的tsfile进行手动merge？iotdb版本1.1.2 。
另外1.1.2 iotdb, 恢复后datanode有如下日志:
`
2024-05-03 06:54:11,273 [pool-15-IoTDB-Compaction_Schedule-root.emsplus-0-1] INFO  o.a.i.d.e.c.s.i.RewriteCrossSpaceCompactionSelector:288 - root.emsplus-0 [Compaction] Total source files: 15 seqFiles, 6 unseqFiles. Candidate source files: 15 seqFiles, 6 unseqFiles. Cannot select any files because they do not meet the conditions or may be occupied by other compaction threads. 2024-05-03 06:54:21,521 [pool-29-IoTDB-LogDispatcher-DataRegion[0]-1] INFO  o.a.i.d.w.n.WALNode$PlanNodeIterator:662 - timeout when waiting for next WAL entry ready, execute rollWALFile. Current search index in wal buffer is 13733221, and next target index is 1.
`
这会导致数据不一致吗？
